### ISSUE TITLE
Razer 16 - Speakers not working

### ISSUE THREAD

**Issue by mrmason** [pop-os/linux#246]
Installed 22.04 LTS to a new Razer 16 laptop and unable to get any sound from the speakers using kernel 6.2.6, and also tried 6.3.0RC4.

Sound output shows :
![image (6)](https://user-images.githubusercontent.com/93157/228510845-e975e430-e997-43c3-b769-faee4368b0be.png)

but there is no output from the inbuilt speakers.

Seeing errors like

```
[  103.975645] snd_hda_codec_realtek ehdaudio0D0: didn't find PCM for DAI Digital Codec DAI
[  103.975650] snd_hda_codec_realtek ehdaudio0D0: ASoC: error at snd_soc_dai_startup on Digital Codec DAI: -22
[  103.975652]  Digital Playback and Capture: ASoC: error at __soc_pcm_open on Digital Playback and Capture: -22
[  103.975654]  HDA Digital: ASoC: error at dpcm_be_dai_startup on HDA Digital: -22
[  103.975655]  HDA Digital: ASoC: error at dpcm_fe_dai_startup on HDA Digital: -22
[  103.975727] snd_hda_codec_realtek ehdaudio0D0: didn't find PCM for DAI Digital Codec DAI
[  103.975729] snd_hda_codec_realtek ehdaudio0D0: ASoC: error at snd_soc_dai_startup on Digital Codec DAI: -22
[  103.975730]  Digital Playback and Capture: ASoC: error at __soc_pcm_open on Digital Playback and Capture: -22
[  103.975731]  HDA Digital: ASoC: error at dpcm_be_dai_startup on HDA Digital: -22
[  103.975732]  HDA Digital: ASoC: error at dpcm_fe_dai_startup on HDA Digital: -22
```

full dmesg attached:

[dmesg.txt](https://github.com/pop-os/linux/files/11099707/dmesg.txt)

**Comment by runboy93** [comment]
Any changes with new Pipewire updates there been meanwhile? Is correct audio device been used?

**Comment by mrmason** [comment]
Seems other distros are having the same issue: https://bbs.archlinux.org/viewtopic.php?id=285121

**Comment by crinavar** [comment]
Hi,
Having the same issue with the same laptop.

**Comment by Goddard** [comment]
same

**Comment by Goddard** [comment]
just an update from a related audio card
https://github.com/thesofproject/linux/issues/4055

**Comment by kstych** [comment]
temp fix available in https://bugzilla.kernel.org/show_bug.cgi?id=207423 , attached works for mt Razer18
[audio.sh.txt](https://github.com/pop-os/linux/files/15366973/audio.sh.txt)

**Comment by j-m-harris** [comment]
@kstych - Thanks for confirming those verbs work for you.

I'm making a kernel patch for the Razer Blade HDA fixups, can you confirm your subsystem ID? You can `cat /sys/class/sound/hwC1D0/subsystem_id` (or whichever card number corresponds for you).

On a RZ09-0483 I've got `0x1a583006`

**Comment by crinavar** [comment]
Hi @j-m-harris 
On my Razer Blade 16 (2023) i've got
```
➜  ~ cat /sys/class/sound/hwC1D0/subsystem_id
0x1a583006```

**Comment by kstych** [comment]
@j-m-harris sure, here goes

Blade 18 - RZ09-0484

```
razer-blade:~$ cat /sys/class/sound/hwC1D0/subsystem_id
0x1a583007
razer-blade:~$ 
```

**Comment by leviport** [comment]
This should probably be reported upstream. System76 is not likely to develop a kernel patch for Razer laptops any time soon.

**Comment by j-m-harris** [comment]
@leviport Yep, will do through https://bugzilla.kernel.org/show_bug.cgi?id=207423

**Comment by v-prgmr** [comment]
@kstych can you point me to the fix?

**Comment by kstych** [comment]
@v-prgmr I have attached audio.sh.txt in my comment , you can download it, and run as sudo

**Comment by v-prgmr** [comment]
@kstych  tried it, but did not fix the issue. Still getting audio out of headphone , but not out of built-in speakers. I am running it on a Razer 16 2024

**Comment by kstych** [comment]
@v-prgmr in my file my device is "/dev/snd/hwC1D0" for Blade-18 , you will need to see what device you have, replace in all lines and try again

**Comment by v-prgmr** [comment]
@kstych how do I find what device I have? I have `hwC0D0,` `hwC1D0,` `hwC1D2` listed in  `/sys/class/sound/` . Should I try each of these?

**Comment by j-m-harris** [comment]
@v-prgmr If you list your devices with `aplay -l` (provided by alsa-utils package, will depend on your distribution) you should be able to find your card and device numbers. Then you will need to replace the hwC{card_num}D{device_num} path accordingly.

For example I am using hwC1D0:

card 1: sofhdadsp [sof-hda-dsp], device 0: HDA Analog (*) []

**Comment by v-prgmr** [comment]
according to `alsa -l`, my device is also `hwC1D0`, is the path /dev/snd/hwc1D0 correct?

**Comment by atty31** [comment]
`cat /sys/class/sound/hwC1D0/subsystem_id
0x1a58300a`
Is there a fix for this issue ?
